src/vscode-bicep/webpack.config.ts

```diff
@@ -1,33 +1,109 @@
-import * as path from 'path';
-import * as CopyPlugin from 'copy-webpack-plugin';
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+import path from "path";
+import webpack from "webpack";
+import CopyPlugin from "copy-webpack-plugin";
+import ForkTsCheckerWebpackPlugin from "fork-ts-checker-webpack-plugin";
 
-module.exports = {
-  target: 'node',
-  entry: './src/extension.ts',
+const extensionConfig: webpack.Configuration = {
+  target: "node",
+  entry: "./src/extension.ts",
+  devtool: "source-map",
   output: {
-    path: path.resolve(__dirname, 'out'),
-    filename: 'extension.js',
+    path: path.resolve(__dirname, "out"),
+    filename: "extension.js",
     libraryTarget: "commonjs2",
+    devtoolModuleFilenameTemplate: "file:///[absolute-resource-path]",
   },
-  devtool: 'source-map',
   externals: {
-    vscode: "commonjs vscode" // the vscode-module is created on-the-fly and must be excluded. Add other modules that cannot be webpack'ed, 📖 -> https://webpack.js.org/configuration/externals/
+    // the vscode-module is created on-the-fly and must be excluded. Add other modules that cannot be webpack'ed, 📖 -> https://webpack.js.org/configuration/externals/
+    vscode: "commonjs vscode",
+    // The following are optional dependencies of vscode-azureextensionui that cannot be resolved.
+    "applicationinsights-native-metrics":
+      "commonjs applicationinsights-native-metrics",
+    "@opentelemetry/tracing": "commonjs @opentelemetry/tracing",
   },
   module: {
-    rules: [{
-      test: /\.ts$/,
-      use: 'ts-loader',
-      exclude: /node_modules/,
-    }]
+    rules: [
+      {
+        test: /\.ts$/,
+        loader: "esbuild-loader",
```

Comment:
Replacing `ts-loader` with `esbuild-loader` reduce by half the bundling time. For now we want to keep `webpack` because we need the plugins, but I'll investigate replacing it completely with [esbuild](https://github.com/evanw/esbuild) (which is about 100x faster).